Mathlib/LinearAlgebra/CliffordAlgebra/Grading.lean

```diff
@@ -165,33 +176,59 @@ theorem evenOdd_induction (n : ZMod 2) {P : ∀ x, x ∈ evenOdd Q n → Prop}
           P (ι Q m₁ * ι Q m₂ * x)
             (zero_add n ▸ SetLike.mul_mem_graded (ι_mul_ι_mem_evenOdd_zero Q m₁ m₂) hx))
     (x : CliffordAlgebra Q) (hx : x ∈ evenOdd Q n) : P x hx := by
-  apply Submodule.iSup_induction' _ _ (hr 0 (Submodule.zero_mem _)) @hadd
+  apply Submodule.iSup_induction' (C := P) _ (hr 0 (Submodule.zero_mem _)) @hadd
   refine' Subtype.rec _
-  simp_rw [Subtype.coe_mk, ZMod.nat_coe_zmod_eq_iff, add_comm n.val]
+  -- porting note: was `simp_rw [Subtype.coe_mk, ZMod.nat_coe_zmod_eq_iff, add_comm n.val]`
+  -- leanprover/lean4#1926 is to blame.
+  dsimp only [Subtype.coe_mk]
+  let hlean1926 : ∀ val : ℕ, ↑val = n ↔ ∃ k, val = 2 * k + ZMod.val n := by
+    intro val
+    simp_rw [ZMod.nat_coe_zmod_eq_iff, add_comm n.val]
+  have := fun val : ℕ => forall_prop_congr'
+    (q := fun property => ∀ x (hx : x ∈ LinearMap.range (ι Q) ^ val),
+      P x (Submodule.mem_iSup_of_mem { val := val, property := property } hx))
+    (hq := fun property => Iff.rfl) (hp := hlean1926 val)
+  simp_rw [this]; clear this
+  -- end porting note
```

Comment:
This is extremely unfortunate, and is a major regression in `simp`'s ability to deal with dependent types.